File: Recursion/RTOH.py
move=0
def TOH(n, src, dst, temp):
    global move
    if n == 1:
        print(f"Move disc {n} from {src} to {dst}")
        move += 1
    else:
        TOH(n-1, src, temp, dst)
        print(f"Move disc {n} from {src} to {dst}")
        move += 1
        TOH(n-1, temp, dst, src)
n = int(input("Enter the number of discs: "))
TOH(n, '1', '3', '2')
print(f"Total moves: {move}")

File: Recursion/RTOH.c
#include<stdio.h>
int move=0;
void TOH(int n, int src, int dst, int temp)
{
    if(n==1){
    printf("Move disc %d from %d to %d\n", n, src, dst);
    move++;}
    else{
        TOH(n-1, src, temp, dst);
        move++;
        printf("Move disc %d from %d to %d\n", n, src, dst);
        TOH(n-1, temp, dst, src);

    }
}
int main()
{
    int n;
    printf("Enter the no. of disc");
    scanf("%d",&n);
    printf("The sequence of moves are:\n");
    TOH(n, 1, 3, 2);
    printf("Total no. of moves= %d\n", move);
    return 0;

}